### ISSUE TITLE
site logo block throws division by zero error when using SVG

### ISSUE THREAD

**Issue by andrewstaffell** [WordPress/gutenberg#36603]
### Description

It's minor but seems to be a result of the attribute width / height being returned as false for SVGs. 

The error is:
```
Warning: Division by zero in /var/www/html/wp-includes/blocks/site-logo.php on line 20
```

That line is:
```
$height = (float) $attributes['width'] / ( (float) $image[1] / (float) $image[2] );
```

### Step-by-step reproduction instructions

1. Install [Safe SVG](https://wordpress.org/plugins/safe-svg/) or any other plugin that allows SVG uploads
2. Insert site logo block
3. Try uploading an SVG
4. View on front end

### Environment info

- WP 5.8.2
- Gutenberg 11.9.1

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by annezazu** [comment]
Hey there! I tried to replicate this by installing the Safe SVG plugin, uploading an SVG image to the site logo block, and viewing it on my site. I had no problem doing so using Gutenberg 11.9.1, WordPress 5.8.2, and the Tove block theme:

https://user-images.githubusercontent.com/26996883/142516403-fde0d922-2572-4b9f-937a-483ae1a5ce8b.mov

Can you provide more context as a result so I can replicate this?

**Comment by andrewstaffell** [comment]
@annezazu do you have debug mode on? (it's a warning rather than an error)

**Comment by annezazu** [comment]
Ah! No :) Just set it in wp-config `define( 'WP_DEBUG', true );`. I still can't replicate.

**Comment by aristath** [comment]
I don't think this is an issue with Gutenberg or WordPress... WP doesn't allow uploading SVGs. If that behavior is altered using a plugin, then that plugin should be properly adding the dimensions to the image props.

**Comment by rvdsteege** [comment]
I stumbled upon the same issue today and understand WordPress does not allow SVGs by default, but wanted to add that since PHP 8 this results in a fatal error instead of a warning. As site logo is usually at the top of the page, this prevents the whole page from loading.

`
Fatal error: Uncaught DivisionByZeroError: Division by zero in /wp-includes/blocks/site-logo.php on line 20
`

The issue only occurs if the image width setting is being used:

<img width="280" alt="Schermafbeelding 2021-12-18 om 18 34 12" src="https://user-images.githubusercontent.com/10371164/146650527-b2dd1b9f-4f53-4f35-9c62-f69c001616b4.png">

**Comment by aristath** [comment]
Please report these issues in the [Safe SVG support forums](https://wordpress.org/support/plugin/safe-svg/). These issues should be fixed in that plugin and not in Gutenberg 👍

**Comment by rvdsteege** [comment]
I'm not using the Safe SVG plugin, but just allowing upload of `image/svg+xml` mime type via the `upload_mimes` filter in my theme instead.

No issues with setting the image width in the default image block with the same SVG upload, so I'm sure the issue is in the site logo block and not in a 3rd party plugin.

**Comment by rvdsteege** [comment]
@annezazu do you still need more info? You should be able to reproduce the issue (PHP warning or fatal error, depending on your PHP version) by changing the image width in the block settings.

**Comment by annezazu** [comment]
Apologies. I'll change the label :)

**Comment by strarsis** [comment]
+1, same issue! Using a SVG logo (which is best practice these days).

A fix is replacing 
```php
if ( empty( $attributes['width'] ) || empty( $image ) ) {
````
with
```php
if ( empty( $attributes['width'] ) || empty( $image ) || $image[1] === 0 || $image[2] === 0 ) {
````
in (Gutenberg plugin/)[`build/block-library/blocks/site-logo.php:17`](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-library/src/site-logo/index.php#L17).

@aristath: IMHO it is good practice anyway to always for a non-zero divisor (with or without SVG support).
There can always happen some issue with determining the image dimensions - and just having a fatal error by an invalid math operation should be avoided.
In this case, `$image[1]` and `$image[2]` are used in a division that itself is used as the divisor in another division,
so both must not be 0 in order to prevent any divisions by zero.

**Comment by hchouhan** [comment]
+1. Having the same issue.

**Comment by strarsis** [comment]
@hchouhan: The SVG support plugins I have tried all don't filter the image dimensions of SVGs, at least incompletely 
(see these posts/issues for plugin [SVG Support](https://wordpress.org/support/topic/specify-svg-width-and-height-dimensions/#post-15255703); [Safe SVG](https://wordpress.org/support/topic/filter-image-dimensions-of-svgs/); [WP SVG images](https://wordpress.org/support/topic/filter-image-dimensions-of-svgs-2/)).

A non-zero fallback would also suffice to fix this issue.

```php
function _svg_dims_fallback_is_svg($image_attachment_id)
{
    $image_mime_type = get_post_mime_type($image_attachment_id);
    return 'image/svg+xml' === $image_mime_type;
}

function svg_dims_fallback($image, $attachment_id, $size, $icon)
{
    if (!_svg_dims_fallback_is_svg($attachment_id)) {
        return $image; // skip non-SVGs
    }

    // set some non-zero defaults to prevent division by zero in Gutenberg site-logo render
    if(!isset($image[1]) or $image[1] === 0) {
        $image[1] = 1;
    }
    if(!isset($image[2]) or $image[2] === 0) {
        $image[2] = 1;
    }

    return $image;
}
add_filter( 'wp_get_attachment_image_src', 'svg_dims_fallback', 10, 4 );

````

The filter has to be set up in a plugin so it runs before the Gutenberg site logo render function.

**Comment by Mamaduka** [comment]
This was fixed via #38808.
